src/Integration/Binding/ConfigProjectBinderFactory.cs

```diff
@@ -23,19 +23,14 @@
 
 namespace SonarLint.VisualStudio.Integration.Binding
 {
-    /// <summary>
-    /// This class was created when refactoring binding to add support for C++ projects.
-    /// It contains shared logic that doesn't currently have a proper home. It shouldn't
-    /// exist (hence the name). It should be removed in the future if/when further
-    /// refactoring is done and a more appropriate location for the responsibilities is
-    /// created.
-    /// </summary>
-    internal static class BindingRefactoringDumpingGround
+    internal class ConfigProjectBinderFactory : IConfigProjectBinderFactory
     {
-        internal /* for testing */ static bool IsProjectLevelBindingRequired(Project project)
+        public IConfigProjectBinder Get(Project project)
         {
-            var languages = ProjectToLanguageMapper.GetAllBindingLanguagesForProject(project);
-            return languages.Contains(Core.Language.VBNET) || languages.Contains(Core.Language.CSharp);
+            var languages = ProjectToLanguageMapper.GetAllBindingLanguagesForProject(project).ToList();
+            var isRoslynProject = languages.Contains(Core.Language.VBNET) || languages.Contains(Core.Language.CSharp);
+
+            return isRoslynProject ? new RoslynConfigProjectBinder() : null;
```

Comment:
As discussed offline, the convention is that `Get` won't return null. What's your plan for this code?

Reply:
I will probably return a no-op, although I'm not a fan of that either.
I'll think about it. Maybe TryGet/TryCreate, not sure yet.